contributing/deployment-tools.txt

```diff
@@ -0,0 +1,113 @@
+OME deployment tools
+====================
+
+This section describes deployment tools supported by the OME team.
+It is primarily designed for the core OME developers who want to
+bring new or upgrade existing prerequisites. The following steps explain
+the connections between basic repositories and the testing workflow.
+
+Prerequisites locations
+-----------------------
+
+List of OME prerequisites is stored in multiple git repositories, each of 
+which is available from several locations.
+
+Infrastructure (Experimental)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Infrastructure is provided to simplify deployment using 
+`Ansible Tower<https://www.ansible.com>`.
+
+The Infrastructure repository is available from:
+
+-  https://github.com/openmicroscopy/infrastructure
+
+
+OMERO-install
+^^^^^^^^^^^^^
+
+OMERO installation scripts are provided to help new users with installing
+OMERO.server for the first time on a clean system, and can be used as
+the basis for more advanced configurations.
+
+The OMERO-install repository is available from:
+
+-  https://github.com/ome/omero-install
+
+
+Testing workflow
+----------------
+
+Devspace (Experimental)
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Continuous integration tools managed by Jenkins CI providing
+an automation framework that runs repeated jobs. The default deployment
+initializes a Jenkins CI master with a predefined set of jobs.
+
+The Devspace repository is available from:
+
+-  https://github.com/openmicroscopy/devspace
+
+Devspace Dockerfiles uses common devslave image. The Devslave repository
+is available from:
+
+-  https://github.com/openmicroscopy/devslave-c7-docker
+
+
+Continuous Delivery (Production)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Production Continuous Delivery (CD) platform managed by
+`Jenkins <http://jenkins-ci.org>`_ available on http://ci.openmicroscopy.org.
+More details about CI-master available on :doc:`continuous-integration`
+
+
+How to add new/upgrade prerequisites
+------------------------------------
+
+When the OME platform requires a new set of prerequisites all the above
+listed repositories may require updates. Depends on the nature of 
+packages developers must consider:
+
+-  adding new Ansible role or complete Ansible playbook to Infrastructure
+-  adding new scripts installing appropriate package and its dependencies
+   to OMERO-install, that includes:
+     - updating Linux and Mac installation scropts
+     - updating documentation autogen
+- deploying Devspace to test OMERO-install scripts, that includes:
+     - adding new Docker container if requires to support additional processes
+     - adjusting predefined Jenkins jobs
+
+After successful testing new prerequisites can be proposed as a permanent
+adjustment to production CD.
+
+.. note::
+
+    Any Python module that is distributed from Linux distro packages
+    must be installed from RPM file. Python modules only available
+    on PyPI should be added as PIP requirement.
```

Comment:
Since non-distribution provided Python prerequisites are actually managed in the source code itself, is it worth adding a section about it at the same level as infrastructure and omero-install mentioning the `requirements` files?